library/std/src/sys/wasi/stdio.rs

```diff
@@ -30,6 +30,14 @@ impl AsFd for Stdin {
     }
 }
 
+#[stable(feature = "io_safety", since = "1.63.0")]
+impl<'a> AsFd for io::StdinLock<'a> {
+    #[inline]
+    fn as_fd(&self) -> BorrowedFd<'_> {
+        unsafe { BorrowedFd::borrow_raw(0) }
```

Comment:
Why exactly is this correct? A safety comment here would be great to have for someone stumbling upon this without having prior knowledge of the API and WASI spec

Reply:
I agree, but note that the implementations of `AsFd` for UNIX don't have such comments either. Nonetheless, added as a suggestion.

Reply:
It's a little tricky. WASI itself doesn't make such a guarantees (Unix doesn't either, for that matter). It's efffectively Rust's std that makes this guarantee/requirement, and for its part, its own API doesn't provide a way to close stdin/stdout/stderr without using `unsafe`. @joshtriplett's added comments look ok for now, though eventually we should add public documentation about this.